File: 136-single-number/single-number.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        
        HashSet<int> hash = new HashSet<int>();
        

        foreach(var n in nums){
            if(!hash.Contains(n)){
            hash.Add(n);
            }else{
            hash.Remove(n);
            }
        }
        return hash.First();
    }
}

File: 136-single-number/single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        std::unordered_set<int> hash;

        for(int n : nums){
            if(!hash.contains(n)){
                hash.insert(n);
            }else{
                hash.erase(n);
            }
        }

        return *hash.begin();

    }
};